src/modules/flow/update/update.c

```diff
@@ -0,0 +1,495 @@
+    mdata->base.progress_port = SOL_FLOW_NODE_TYPE_UPDATE_CHECK__OUT__PROGRESS;
+
+    if (opts->url && strlen(opts->url) > 0) {
+        mdata->url = strdup(opts->url);
+        SOL_NULL_CHECK(mdata->url, -ENOMEM);
+    }
+
+    return 0;
+}
+
+static void
+check_close(struct sol_flow_node *node, void *data)
+{
+    struct check_data *mdata = data;
+
+    if (mdata->base.handle)
+        sol_update_cancel(mdata->base.handle);
+
+    free(mdata->url);
+}
+
+static int
+cancel_check_process(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    struct check_data *mdata = data;
+
+    if (mdata->base.handle) {
+        if (!sol_update_cancel(mdata->base.handle))
+            sol_flow_send_error_packet(node, EINVAL,
+                "Could not cancel check process");
+        else
+            mdata->base.handle = NULL;
+    } else {
+        SOL_WRN("No current check in process, ignoring request to cancel");
+    }
+
+    return 0;
+}
+
+static struct sol_blob *
+serialize_info(const struct sol_update_info *response)
+{
+    struct sol_buffer buffer = SOL_BUFFER_INIT_EMPTY;
+    struct sol_blob *blob;
+    size_t size;
+    void *v;
+    int r;
+
+    r = sol_buffer_append_slice(&buffer, (const struct sol_str_slice)SOL_STR_SLICE_LITERAL("{\"version\":"));
+    SOL_INT_CHECK_GOTO(r, < 0, err);
+    r = sol_json_serialize_string(&buffer, response->version);
+    SOL_INT_CHECK_GOTO(r, < 0, err);
+
+    r = sol_buffer_append_slice(&buffer, (const struct sol_str_slice)SOL_STR_SLICE_LITERAL(",\"url\":"));
+    SOL_INT_CHECK_GOTO(r, < 0, err);
+    r = sol_json_serialize_string(&buffer, response->url);
+    SOL_INT_CHECK_GOTO(r, < 0, err);
+
+    r = sol_buffer_append_slice(&buffer, (const struct sol_str_slice)SOL_STR_SLICE_LITERAL(",\"hash\":"));
+    SOL_INT_CHECK_GOTO(r, < 0, err);
+    r = sol_json_serialize_string(&buffer, response->hash);
+    SOL_INT_CHECK_GOTO(r, < 0, err);
+
+    r = sol_buffer_append_slice(&buffer, (const struct sol_str_slice)SOL_STR_SLICE_LITERAL(",\"hash-algorithm\":"));
+    SOL_INT_CHECK_GOTO(r, < 0, err);
+    r = sol_json_serialize_string(&buffer, response->hash_algorithm);
+    SOL_INT_CHECK_GOTO(r, < 0, err);
+
+    r = sol_buffer_append_char(&buffer, '}');
+    SOL_INT_CHECK_GOTO(r, < 0, err);
+
+    v = sol_buffer_steal(&buffer, &size);
+    blob = sol_blob_new(SOL_BLOB_TYPE_DEFAULT, NULL, v, size);
+    SOL_NULL_CHECK_GOTO(blob, err);
+
+    return blob;
+err:
+    sol_buffer_fini(&buffer);
+
+    return NULL;
+}
+
+static void
+check_cb(void *data, int status, const struct sol_update_info *response)
+{
+    struct base_data *mdata = data;
+    struct sol_blob *blob;
+
+    if (status < 0) {
+        sol_flow_send_error_packet(mdata->node, -status,
+            "Error while checking for updates");
+        goto end;
+    }
+
+    /* TODO maybe a custom packet instead of JSON? Maybe sol_update_info have
+     * the raw json? */
```

Comment:
you may rely on composed-packet with basic types in a certain order. Then you document the order so people can use composed type meta-type to generate extractors. Talk to @cabelitos